mlflow_elasticsearchstore/elasticsearch_store.py

```diff
@@ -180,28 +222,93 @@ def set_tag(self, run_id: str, tag: RunTag) -> None:
             .filter('nested', inner_hits={"size": 100}, path="metrics",
-                    query=Q('term', metrics__key=metric_key)).source("false").execute()
-        return [self._hit_to_mlflow_metric(m._source) for m in
+                    query=Q('term', metrics__key=metric_key)).source(False).execute()
+        return [self._hit_to_mlflow_metric(m["_source"]) for m in
                 response["hits"]["hits"][0].inner_hits.metrics.hits.hits]
 
+    def list_all_columns(self, experiment_id: str, run_view_type: str) -> Columns:
+        stages = LifecycleStage.view_type_to_stages(run_view_type)
+        s = Search(index="mlflow-runs").filter("match", experiment_id=experiment_id)\
+            .filter("terms", lifecycle_stage=stages)
+        s.aggs.bucket('latest_metrics', 'nested', path="latest_metrics") \
+            .bucket("latest_metrics_keys", "terms", field="latest_metrics.key")
+        s.aggs.bucket('params', 'nested', path="params") \
+            .bucket("params_keys", "terms", field="params.key")
+        s.aggs.bucket('tags', 'nested', path="tags") \
+            .bucket("tags_keys", "terms", field="tags.key")
+        response = s.execute()
+        metrics = [m.key for m in response.aggregations.latest_metrics.latest_metrics_keys.buckets]
+        params = [p.key for p in response.aggregations.params.params_keys.buckets]
+        tags = [t.key for t in response.aggregations.tags.tags_keys.buckets]
+        return Columns(metrics=metrics, params=params, tags=tags)
+
+    def _build_elasticsearch_query(self, parsed_filters: List[dict], s: Search) -> Search:
+        type_dict = {"metric": "latest_metrics", "parameter": "params", "tag": "tags"}
+        for search_filter in parsed_filters:
+            key_type = search_filter.get('type')
+            key_name = search_filter.get('key')
+            value = search_filter.get('value')
+            comparator = search_filter.get('comparator').upper()
+            if key_type == "parameter":
+                query = Q("term", params__key=key_name) & Q("term", params__value=value)
+            elif key_type == "tag":
+                query = Q("term", tags__key=key_name) & Q("term", tags__value=value)
+            elif key_type == "metric":
+                query = Q("term", latest_metrics__key=key_name)
+                if comparator == "=":
+                    query = query & Q("term", latest_metrics__value=value)
+                elif comparator == ">":
+                    query = query & Q("range", latest_metrics__value={'gt': value})
+                elif comparator == ">=":
+                    query = query & Q("range", latest_metrics__value={'gte': value})
+                elif comparator == "<":
+                    query = query & Q("range", latest_metrics__value={'lt': value})
+                elif comparator == "<=":
+                    query = query & Q("range", latest_metrics__value={'lte': value})
```

Comment:
You can do better than all these if, you can create a dictionnary that maps each comparator to what you need to put in your elasticsearch query